src/Arcus.WebApi.Logging.AzureFunctions/AzureFunctionsRequestTrackingMiddleware.cs

```diff
@@ -72,13 +73,8 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
 
         private static async Task<HttpRequestData> EnableHttpRequestBufferingAsync(FunctionContext context)
         {
-            BindingMetadata bindingMetadata = context.FunctionDefinition.InputBindings.Values.FirstOrDefault(a => a.Type == "httpTrigger");
-            if (bindingMetadata is null)
-            {
-                throw new InvalidOperationException(
+            BindingMetadata bindingMetadata = context.FunctionDefinition.InputBindings.Values.FirstOrDefault(a => a.Type == "httpTrigger") ?? throw new InvalidOperationException(
```

Comment:
This chnge is a bit unrelated to the guard.net stuff, nd I believe it is not an improvement readability-wise.

Reply:
I agree with the readability comment.
If we would expect that the collection always contains at least one item that complies to the filter statement, we could also use `First` instead of `FirstOrDefault` .  Drawback however, is that we we then have to live with the exception that is thrown by `First` if no element is found, which is less clear then the exception we throw ourselves here

Reply:
I've reverted the change.